File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/callback/HttpCallback.java
package com.github.kongpf8848.rxhttp.callback;

import android.util.Log;

import com.github.kongpf8848.rxhttp.util.TypeUtil;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;

public abstract class HttpCallback<T> implements BaseCallback<T> {
    private Type type;

    public HttpCallback() {
        this.type = TypeUtil.getType(getClass());
    }

    public HttpCallback(Type type){
        this.type=type;
    }

    public Type getType() {
        return type;
    }

}

File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/callback/HttpCallback.kt
package com.github.kongpf8848.rxhttp.callback

import com.github.kongpf8848.rxhttp.util.TypeUtil
import java.lang.reflect.Type

abstract class HttpCallback<T> {

    var type: Type

    constructor() {
        type = TypeUtil.getType(javaClass)
    }

    constructor(type: Type) {
        this.type = type
    }

    /**
     * http请求开始时回调
     */
    open fun onStart(){}

    /**
     * http请求成功时回调
     */
    abstract fun onNext(response: T?)

    /**
     * http请求失败时回调
     */
    abstract fun onError(e: Throwable?)

    /**
     * http请求完成时回调
     */
    open fun onComplete(){}

    /**
     * 上传下载进度回调
     * @param readBytes
     * @param totalBytes
     */
    open fun onProgress(readBytes: Long, totalBytes: Long){}
}